### PR TITLE
Reword contributing guide

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Updates the contributing guide. This makes the instructions a bit more practical by listing out the common gradle tasks used during local development.

### CODE REVIEW

**Comment by tomlongridge** [comment]
I find this structure a little hard to follow. Can I suggest we keep a Testing page with the Unit, Integration and End-to-End test instructions? Then the Contributing Guide will link down separately to Developing (inc Static Analysis, Building Locally, Custom ABIs), Testing (inc Unit, Integration, e2e and Example App) and Release subpages.

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+```shell
+./gradlew detekt
+```
+
+Warnings can be suppressed by running the following task:
+
+```shell
+./gradlew detektBaseline
+```
```
Comment (kstenerud) [inline DEVELOPMENT.md:67]: Should probably add a warning that this permanently suppresses ALL outstanding detekt warnings currently in the project, including anything the developer hasn't investigated yet! And that this modifies `detekt-baseline.xml`, which must then be committed.

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+You should open `examples/sdk-app-example` as a separate Android Studio project and [run the app](https://developer.android.com/training/basics/firstapp/running-app). This will require a connected device or emulator.
+
+The app **does not use the local development version of Bugsnag**. To test local development changes, you will need to build the library locally.
+
+## Building the Library locally
+
+The following command assembles bugsnag-android for local use:
+
+```shell
+./gradlew assembleRelease publishToMavenLocal -PVERSION_NAME=9.9.9
```
Comment (kstenerud) [inline DEVELOPMENT.md:106]: Does setting `VERSION_NAME` replace the two steps of modifying `gradle.properties` and `sdk-app-example/build.gradle` to have the same large, unassigned bugsnag-android version numbers?

**Review by kstenerud — CHANGES_REQUESTED** [review]

File: DEVELOPMENT.md (hunk 2 above)
Comment (fractalwrench) [reply DEVELOPMENT.md:106]: Updated to state that the example app's version number must be updated too

File: DEVELOPMENT.md (hunk 1 above)
Comment (fractalwrench) [reply DEVELOPMENT.md:67]: Updated to state that this file will be written to.

**Comment by fractalwrench** [comment]
Updated according to feedback and restructured how the guides are organised.

**Review by kstenerud — APPROVED** [review]
